src/model/buffer.rs

```diff
@@ -472,6 +474,70 @@ impl Buffer {
         vec![left_str, right_str]
     }
 
+    pub fn as_tagged_text(&self) -> TaggedText {
+        // construct a serialised string by iterating through the nodes
+        //  and copying the contents that they refer to based on
+        //  their indices and offsets and which string they refer to
+
+        if self.node_list.is_empty() {
+            return TaggedText::new(String::new(), vec![TextTag::new(Tag::Cursor, 0, 1)]);
+        }
+        let mut tags: Vec<TextTag> = Vec::new();
+
+        let left_iter = self.node_list.iter_until_curr();
+        let right_iter = self.node_list.iter_from_after_curr();
+        let mut left_str = String::new();
+        let mut right_str = String::new();
```

Comment:
Is `right_str` going to be used at all? It seems that whatever is from the contents of the `right_iter` is appended on to `left_str` directly

Reply:
eh right this was carried over from the other implementation, will remove when i get to it